localstack/testing/pytest/fixtures.py

```diff
@@ -1378,6 +1378,27 @@ def _create_function():
             LOG.debug(f"Unable to delete log group {log_group_name} in cleanup")
 
 
+@pytest.fixture
+def upload_file(s3_create_bucket, s3_client):
+    def _upload_file(file_path: str, bucket_name: str = None):
+        if not bucket_name:
+            bucket_name = s3_create_bucket()
+        key = f"file-{short_uid()}"
+
+        s3_client.upload_file(
+            file_path,
+            Bucket=bucket_name,
+            Key=key,
+        )
+
+        domain = "amazonaws.com" if is_aws_cloud() else "localhost.localstack.cloud:4566"
+        url = f"https://{bucket_name}.s3.{domain}/{key}"
+
+        return {"Bucket": bucket_name, "Key": key, "Url": url}
+
+    yield _upload_file
```

Comment:
I'd discourage the usage of fixtures for this. We don't really benefit from this being a fixture (compared to it just being a stateless test utility function) since there's no cleanup necessary.